### PR TITLE
Restructure `Package.swift`

### PR DESCRIPTION
**Pull request by FranzBusch**
This might be a controversial PR! 🍿

# Motivation
It always bothered me how our `Package.swift` is structured. First, we had a huge list of `targets` at the top that distracted from the `products`. The `products` are the important part for our adopters since those are the actual modules they can use. Secondly, the formatting of the targets made it hard to follow and resulted in unnecessary merge conflicts.

# Modification
This PR moves the targets back to the bottom and splits all the various array elements into separate elements.

# Result
Easier to understand `Package.swift` and less merge conflicts.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -13,115 +13,10 @@
...
-    .testTarget(name: "NIOWebSocketTests",
-                dependencies: ["NIOCore", "NIOEmbedded", "NIOWebSocket"]),
-    .testTarget(name: "NIOTestUtilsTests",
-                dependencies: ["NIOTestUtils", "NIOCore", "NIOEmbedded", "NIOPosix"]),
-    .testTarget(name: "NIOFoundationCompatTests",
-                dependencies: ["NIOCore", "NIOFoundationCompat"]),
-    .testTarget(name: "NIOTests",
-                dependencies: ["NIO"]),
-]
+import PackageDescription
```
Comment (dnadoba) [inline Package.swift:19]: The `import` should be at the top before the dependency definitions.

File: Package.swift [hunk 2]
```diff
@@ -143,5 +38,351 @@ let package = Package(
         .package(url: "https://github.com/apple/swift-collections.git", from: "1.0.2"),
         .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
     ],
-    targets: targets
+    targets: [
```
Comment (glbrntt) [inline Package.swift:41]: I think it would be useful to group targets into 'blocks', e.g. "modules", "examples", and "tests" and separate each with a `// MARK:` or similar.

Comment (FranzBusch) [reply Package.swift:42]: Added `// MARK:`s and grouped them

File: Package.swift [hunk 3]
```diff
@@ -143,5 +39,359 @@ let package = Package(
...
+        // MARK: Targets
```
Comment (0xTim) [inline Package.swift:43]:
```suggestion
        // MARK: - Targets
```
Makes use of Xcode's formatting for splitting up stuff

**Review by Lukasa — APPROVED** [review]
Sure, why not.

**Comment by FranzBusch** [comment]
Merging over the known allocation regression.
